src/dune_rules/gen_rules.ml

```diff
@@ -146,6 +146,82 @@ end = struct
       | true ->
         let+ () = Mdx.gen_rules ~sctx ~dir ~scope ~expander mdx in
         empty_none)
+    | Melange mel ->
+      let melange_stanza_dir = dir in
+      let all_libs_compile_info =
+        let dune_version = Scope.project scope |> Dune_project.dune_version in
+        let pps = [] in
+        Lib.DB.resolve_user_written_deps_for_exes (Scope.libs scope)
+          [ (mel.loc, mel.target) ]
+          mel.libraries ~pps ~dune_version
+      in
+      let lib_cctx ~sctx ~obj_dir ~modules ~expander ~scope ~compile_info =
+        let flags = Ocaml_flags.empty in
+        let vimpl =
+          (* original impl in Lib_rules uses Virtual_rules.impl, will this break with virtual libs? *)
+          None
+        in
+        let modules = Vimpl.impl_modules vimpl modules in
+        let requires_compile = Lib.Compile.direct_requires compile_info in
+        let requires_link = Lib.Compile.requires_link compile_info in
+        let package = None in
+        let js_of_ocaml = None in
+        let melange = (melange_stanza_dir, mel.target) in
+        (* modes and pp are not passed, not sure if this will cause issues *)
+        Compilation_context.create () ~super_context:sctx ~expander ~scope
+          ~obj_dir ~modules ~flags ~requires_compile ~requires_link
+          ~opaque:Inherit_from_settings ~js_of_ocaml ~package ~melange ?vimpl
+      in
+      let rules (mel : Melange_stanza.t) ~sctx ~melange_stanza_dir ~scope
+          ~expander =
+        let open Memo.O in
+        let* libs = Lib.Compile.direct_requires all_libs_compile_info in
```

Comment:
This should be the linking closure (`requires_link`)

Reply:
Nice, i was bothered because one had to list transient lib deps manually in the `melange.emit` stanza, but this change fixed it. See tests changes in https://github.com/ocaml/dune/pull/6349/commits/270b83e480db47432100fbae7035570bcfed51f4.